packages/account/src/Components/trading-assessment/trading-assessment-form.jsx

```diff
@@ -0,0 +1,201 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Formik, Form } from 'formik';
+import { Button, Modal, FormSubmitButton, Text, Icon } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import { localize, Localize } from '@deriv/translations';
+import TradingAssessmentRadioButton from './trading-assessment-radio-buttons.jsx';
+import TradingAssessmentDropdown from './trading-assessment-dropdown.jsx';
+
+const TradingAssessmentForm = ({
+    assessment_questions,
+    class_name,
+    form_value,
+    onSubmit,
+    onCancel,
+    is_header_navigation,
+    should_move_to_next,
+}) => {
+    const [is_next_button_enabled, setIsNextButtonEnabled] = React.useState(false);
+    const [current_question_details, setCurrentQuestionDetails] = React.useState({
+        current_question_index: 0,
+        current_question: {},
+    });
+    const [form_data, setFormData] = React.useState({});
+
+    const last_question_index = assessment_questions.length - 1;
+
+    React.useEffect(() => {
+        setCurrentQuestionDetails(prevState => ({
+            ...prevState,
+            current_question: assessment_questions[prevState.current_question_index],
+        }));
+        setFormData(form_value);
+    }, []);
+
+    React.useEffect(() => {
+        if (should_move_to_next) displayNextPage();
+    }, [should_move_to_next]);
```

Comment:
Don’t really understand this part of code... should_move_to_next is a boolean. and this variable is in dependency array. so displayNextPage will not be called if should_move_to_next is false, but in the same time, when it is always true, it also will not be called, as the variable in dependency list will not be changed. 
Don’t understand the goal of using here useEffect. and the logic on the whole. Could please you explain?

Reply:
They want to listen to changes on `should_move_to_next ` and proceed to the next step whenever it changes to `true`

Reply:
Will explain u seperately